docs/src/content/docs/commands/ZRANK.md

```diff
@@ -0,0 +1,104 @@
+---
+title: ZRANK
+description: Documentation for the DiceDB command ZRANK
+---
+
+The `ZRANK` command is used to determine the rank of a member in a sorted set stored at a specific key in DiceDB. The rank is based on the score of the member, with the member with the lowest score having rank 0.
+
+## Syntax
+
+```
+ZRANK key member [WITHSCORE]
+```
+
+## Parameters
+
+| Parameter   | Description                                                                 | Type   | Required |
+|-------------|-----------------------------------------------------------------------------|--------|----------|
+| `key`       | The key of the sorted set.                                                  | String | Yes      |
+| `member`    | The member whose rank is to be determined.                                 | String | Yes      |
+| `WITHSCORE` | (Optional) If provided, the command will also return the score of the member.| String | No       |
+
+## Return Value
+
+- **Integer**: The rank of the member in the sorted set, with the scores ordered from low to high. The rank is 0-based.
+- **Array**: If the `WITHSCORE` option is used, returns an array containing the rank and the score of the member.
+- **`nil`**: If the member does not exist in the sorted set or the key does not exist.
+
+## Behaviour
+
+When the `ZRANK` command is executed:
+
+1. **Key Verification**:
+   - DiceDB checks if the specified key exists.
+   - If the key does not exist, `nil` is returned.
+   
+2. **Type Verification**:
+   - If the key exists, DiceDB verifies that it is associated with a sorted set.
+   - If the key is not a sorted set, a `WRONGTYPE` error is returned.
+   
+3. **Member Search**:
+   - DiceDB searches for the specified member within the sorted set.
+   - If the member exists, its rank is determined based on its score.
+   
+4. **`WITHSCORE` Option**:
+   - If the `WITHSCORE` option is provided and the member exists, DiceDB returns both the rank and the score of the member.
+   
+5. **Non-Existent Member**:
```

Comment:
Please check these files for reference and format for this section: docs/command_docs_template.md and docs/sample_command_docs.md